storage/cloud.py

```diff
@@ -696,6 +696,28 @@ def cancel_chunked_upload(self, uuid, storage_metadata):
         for chunk in self._chunk_list_from_metadata(storage_metadata):
             self.remove(chunk.path)
 
+    def clean_partial_uploads(self, deletion_date_threshold):
+        self._initialize_cloud_conn()
+        path = self._init_path("uploads")
+        paginator = self.get_cloud_conn().get_paginator(self._list_object_version)
+        for page in paginator.paginate(Bucket=self._bucket_name, Prefix=path):
+            for obj_info in page.get("Contents", []):
+                if obj_info["LastModified"] <= datetime.now(timezone.utc) - deletion_date_threshold:
+                    obj = self.get_cloud_bucket().Object(obj_info["Key"])
+                    try:
+                        obj.load()
```

Comment:
This will check if the blob in the uploads folder has existed longer than `DELETION_DATE_THRESHOLD`, to prevent the clean up from deleting blobs in the uploads folder that are currently being used as a multi-part upload.

Reply:
How does that work, since delete called immediately after?

Reply:
To clarify, which call are you referring to? the `obj.load()`?

Reply:
Yes, isn't that just doing a HEAD request?

Reply:
Correct, the remove logic is [ported over from the remove method](https://github.com/bcaton85/quay/blob/gc-uploads/storage/cloud.py#L424-L429) which also includes the `obj.load()`. It's used to check if the object exists through [the info provided in the metadata, this stackoverflow has more info.](https://stackoverflow.com/questions/33842944/check-if-a-key-exists-in-a-bucket-in-s3-using-boto3)

Reply:
[It's only used to adjust the error output to make it a bit cleaner.](https://github.com/bcaton85/quay/blob/gc-uploads/storage/cloud.py#L714-L721)